Feature/watch party

smfulton opened the pull request:
Hello All.

I have watchparty functional but will need some adjustments for final product. (Outside voting and movies this is mostly CSS, maybe HTML structure)

- Users can create a watch party and add movies (test data in JS), add members, and name the watch party.
- Users can view their watch party list, edit (Leader has options members do not), leave/delete.
- Temporary option in place for voting on movies but not functional. (Only what was discussed with Cody & Bekah)
- Individual movie detail pages have a button for adding the movie to a watch party from the movie detail page. (must be authenticated to see)
- Adjusted account dropdown and main nav to include watch party.

I would suggest dropping all tables except for roles for this update.
Please review and let me know if you have questions.

smfulton committed:
Working to add groups to profiles.
Push needed before Chicago trip to ensure Mac is up to date.
Will try to work on between today and 30th. If not time is available after.

smfulton committed:
Created watchparty and connected hosted and joined watchparties to the profile model. Actively building create so the host can add members on creation. Create will use a RestController for searching profiles that are not hidden. End goal is to prioritize friends and show hidden friends. Need to add friends to profile to allow this.

smfulton committed:
Created watchparty and connected hosted and joined watchparties to the profile model. Actively building create so the host can add members on creation. Create will use a RestController for searching profiles that are not hidden. End goal is to prioritize friends and show hidden friends. Need to add friends to profile to allow this.

smfulton committed:
watchlist index uses tabs to display hosted watch parties and joined watch parties. Information is populated from Profile.java.
Added create to WatchParty. Needs edit and delete before additions.
Added SearchRestController and a method for searching profiles by name.
Uses case-insensitive partial match search for profiles.
JS on watchparty/create to access restcontroller and displays profile entries as options in a select element within the create form.

Option for edit and delete still coming. Will try best to complete today.

smfulton committed:
Delete watchlist is complete. Single Watchlist details page is more than expected. I will continue working on this tomorrow. I will also tackle editing tomorrow and try to PR asap

smfulton committed:
WatchParty edit and delete has been added. Some things in edit are locked to the host.
Users can add movies to WatchParty from movie-details page if they are logged in and select a watch party.
Duplicate watchparty possible if user adds their own profile to the member list.

Temp code in place for votes and movie search.

bekahkozinski reviewed (commented):
Great work as always! I didn't have any questions because I felt like I was well informed prior to the PR since we had a couple meetings!
Thanks again for being consistently awesome!

smfulton committed:
WatchParty to Profile connection is fixed.
Had to adjust relationship annotations to resolve needing to modify Profile and WatchParty rather than just WatchParty.

smfulton committed:
Left a debug sout in WatchPartyController which is now removed.

CodyA1117 reviewed (approved):
Great work with everything!  I love the way its built, and I will get the voting system added to Watchparty as soon as possible!

bekahkozinski commented:
APPROVED <3
